xds/internal/httpfilter/rbac/rbac.go

```diff
@@ -106,7 +107,42 @@ func parseConfig(rbacCfg *rpb.RBAC) (httpfilter.FilterConfig, error) {
 			}
 		}
 	}
-	return config{config: rbacCfg}, nil
+
+	// "Envoy aliases :authority and Host in its header map implementation, so
+	// they should be treated equivalent for the RBAC matchers; there must be no
+	// behavior change depending on which of the two header names is used in the
+	// RBAC policy." - A41. Loop through config's principals and policies, change
+	// any header matcher with value "host" to :authority", as that is what
+	// grpc-go shifts both headers to in transport layer.
+	for _, policy := range rbacCfg.GetRules().GetPolicies() {
+		for _, principal := range policy.Principals {
+			if principal.GetHeader() != nil {
```

Comment:
This `nil` check is unnecessary.  `principal.GetHeader().GetName()` will return `""` if `Header` is `nil`.  And below.

Reply:
Changed.

Reply:
Did you forget to push a commit?

Reply:
Oh, I deleted the nil check above, but not here. Deleted, and also working on not persisting config.